### PR TITLE
add debug flag for zli commands

### PR DESCRIPTION
**Pull request by aokirisaki**
Signed-off-by: Lisca Ana-Roberta <ana.kagome@yahoo.com>

**What type of PR is this?**

**Which issue does this PR fix**:
#593 

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by aokirisaki** [comment]
This is an output example:
[9/14 5:14 PM] Lisca, Ana-Roberta (DXC Luxoft)
Searching... 🌍GET   http://127.0.0.1:8080/v2/_catalogGET http://127.0.0.1:8080/v2/_catalog Status code  200GET   http://127.0.0.1:8080/v2/zot-test/tags/listGET   http://127.0.0.1:8080/v2/golang/tags/listGET   http://127.0.0.1:8080/v2/hello-artifact/tags/listGET   http://127.0.0.1:8080/v2/test/data/zot-test/blobs/sha256/2d473b07cdd5f0912cd6f1a703352c82b512407db6b05b43f2553732b55df3bc/tags/listGET   http://127.0.0.1:8080/v2/busybox/tags/listGET   http://127.0.0.1:8080/v2/test-regclient/tags/listGET   http://127.0.0.1:8080/v2/node/tags/listGET   http://127.0.0.1:8080/v2/python/tags/listGET   http://127.0.0.1:8080/v2/hello-world/tags/listGET   http://127.0.0.1:8080/v2/zot-chart/zot/tags/listGET   http://127.0.0.1:8080/v2/zot-cve-test/tags/listGET http://127.0.0.1:8080/v2/hello-world/tags/list Status code  200GET http://127.0.0.1:8080/v2/zot-test/tags/list Status code  200GET http://127.0.0.1:8080/v2/zot-cve-test/tags/list Status code  200GET http://127.0.0.1:8080/v2/hello-artifact/tags/list Status code  200GET http://127.0.0.1:8080/v2/test-regclient/tags/list Status code  200GET http://127.0.0.1:8080/v2/busybox/tags/list Status code  200GET http://127.0.0.1:8080/v2/node/tags/list Status code  200GET http://127.0.0.1:8080/v2/golang/tags/list Status code  200GET http://127.0.0.1:8080/v2/test/data/zot-test/blobs/sha256/2d473b07cdd5f0912cd6f1a703352c82b512407db6b05b43f2553732b55df3bc/tags/list Status code  200GET   http://127.0.0.1:8080/v2/hello-artifact/manifests/v2GET http://127.0.0.1:8080/v2/python/tags/list Status code  200GET http://127.0.0.1:8080/v2/zot-chart/zot/tags/list Status code  200GET http://127.0.0.1:8080/v2/hello-artifact/manifests/v2 Status code  200IMAGE NAME                        TAG                       DIGEST        SIZE      hello-artifact                    v2                        74470033      12B       GET   http://127.0.0.1:8080/v2/test-regclient/manifests/1.0.0GET http://127.0.0.1:8080/v2/test-regclient/manifests/1.0.0 Status code  200test-regclient                    1.0.0                     d8be8fb7      76MB      GET   http://127.0.0.1:8080/v2/zot-cve-test/manifests/0.0.1GET http://127.0.0.1:8080/v2/zot-cve-test/manifests/0.0.1 Status code  200zot-cve-test                      0.0.1                     98f02d9e      75MB      GET   http://127.0.0.1:8080/v2/test-regclient/manifests/latestGET http://127.0.0.1:8080/v2/test-regclient/manifests/latest Status code  200test-regclient                    latest                    d66fe387      76MB      GET   http://127.0.0.1:8080/v2/busybox/manifests/latestGET http://127.0.0.1:8080/v2/busybox/manifests/latest Status code  200busybox                           latest                    b9590ac9      773kB     GET   http://127.0.0.1:8080/v2/busybox/manifests/1.0.1GET http://127.0.0.1:8080/v2/busybox/manifests/1.0.1 Status code  200busybox                           1.0.1                     2ac88360      773kB

**Comment by aokirisaki** [comment]
This is an output example:
Searching... 🌍GET   http://127.0.0.1:8080/v2/_catalogGET http://127.0.0.1:8080/v2/_catalog Status code  200GET   http://127.0.0.1:8080/v2/zot-test/tags/listGET   http://127.0.0.1:8080/v2/golang/tags/listGET   http://127.0.0.1:8080/v2/hello-artifact/tags/listGET   http://127.0.0.1:8080/v2/test/data/zot-test/blobs/sha256/2d473b07cdd5f0912cd6f1a703352c82b512407db6b05b43f2553732b55df3bc/tags/listGET   http://127.0.0.1:8080/v2/busybox/tags/listGET   http://127.0.0.1:8080/v2/test-regclient/tags/listGET   http://127.0.0.1:8080/v2/node/tags/listGET   http://127.0.0.1:8080/v2/python/tags/listGET   http://127.0.0.1:8080/v2/hello-world/tags/listGET   http://127.0.0.1:8080/v2/zot-chart/zot/tags/listGET   http://127.0.0.1:8080/v2/zot-cve-test/tags/listGET http://127.0.0.1:8080/v2/hello-world/tags/list Status code  200GET http://127.0.0.1:8080/v2/zot-test/tags/list Status code  200GET http://127.0.0.1:8080/v2/zot-cve-test/tags/list Status code  200GET http://127.0.0.1:8080/v2/hello-artifact/tags/list Status code  200GET http://127.0.0.1:8080/v2/test-regclient/tags/list Status code  200GET http://127.0.0.1:8080/v2/busybox/tags/list Status code  200GET http://127.0.0.1:8080/v2/node/tags/list Status code  200GET http://127.0.0.1:8080/v2/golang/tags/list Status code  200GET http://127.0.0.1:8080/v2/test/data/zot-test/blobs/sha256/2d473b07cdd5f0912cd6f1a703352c82b512407db6b05b43f2553732b55df3bc/tags/list Status code  200GET   http://127.0.0.1:8080/v2/hello-artifact/manifests/v2GET http://127.0.0.1:8080/v2/python/tags/list Status code  200GET http://127.0.0.1:8080/v2/zot-chart/zot/tags/list Status code  200GET http://127.0.0.1:8080/v2/hello-artifact/manifests/v2 Status code  200IMAGE NAME                        TAG                       DIGEST        SIZE      hello-artifact                    v2                        74470033      12B       GET   http://127.0.0.1:8080/v2/test-regclient/manifests/1.0.0GET http://127.0.0.1:8080/v2/test-regclient/manifests/1.0.0 Status code  200test-regclient                    1.0.0                     d8be8fb7      76MB      GET   http://127.0.0.1:8080/v2/zot-cve-test/manifests/0.0.1GET http://127.0.0.1:8080/v2/zot-cve-test/manifests/0.0.1 Status code  200zot-cve-test                      0.0.1                     98f02d9e      75MB      GET   http://127.0.0.1:8080/v2/test-regclient/manifests/latestGET http://127.0.0.1:8080/v2/test-regclient/manifests/latest Status code  200test-regclient                    latest                    d66fe387      76MB      GET   http://127.0.0.1:8080/v2/busybox/manifests/latestGET http://127.0.0.1:8080/v2/busybox/manifests/latest Status code  200busybox                           latest                    b9590ac9      773kB     GET   http://127.0.0.1:8080/v2/busybox/manifests/1.0.1GET http://127.0.0.1:8080/v2/busybox/manifests/1.0.1 Status code  200busybox                           1.0.1                     2ac88360      773kB

File: pkg/cli/client.go [hunk 1]
```diff
@@ -119,11 +122,19 @@ func doHTTPRequest(req *http.Request, verifyTLS bool, resultsPtr interface{}) (h
...
+		fmt.Fprintln(configWriter, req.Method, " ", req.URL)
+	}
+
 	resp, err := httpClient.Do(req)
 	if err != nil {
 		return nil, err
 	}
 
+	if debug {
+		fmt.Fprintln(configWriter, req.Method, req.URL, "Status code ", resp.StatusCode)
```
Comment (rchincha) [inline pkg/cli/client.go:135]: Call it a single word "status".
Print request/response headers also.

Comment (aokirisaki) [reply pkg/cli/client.go]: I have made the changes

File: pkg/cli/client.go [hunk 2]
```diff
@@ -119,11 +122,19 @@ func doHTTPRequest(req *http.Request, verifyTLS bool, resultsPtr interface{}) (h
 
 	httpClientLock.Unlock()
 
+	if debug {
+		fmt.Fprintln(configWriter, req.Method, " ", req.URL, "request header ", req.Header)
```
Comment (andaaron) [inline pkg/cli/client.go:126]: Please add a `[debug] ` prefix to these prints, or make sure they are indented

**Comment by aokirisaki** [comment]
Output example:

 ``` 
[debug]  GET   http://127.0.0.1:5000/v2/_zot/ext/search?query=%7BImageList%28repo%3A+%22%22%29+%7B%0A%09%09%09%09%09%09%09%09%09RepoName+Tag+Digest+ConfigDigest+Size+Layers+%7BSize+Digest%7D%7D%0A%09%09%09%09%09%09%09++%7D [request header]  map[Authorization:[Basic Og==] Content-Type:[application/json]]
[debug]  GET http://127.0.0.1:5000/v2/_zot/ext/search?query=%7BImageList%28repo%3A+%22%22%29+%7B%0A%09%09%09%09%09%09%09%09%09RepoName+Tag+Digest+ConfigDigest+Size+Layers+%7BSize+Digest%7D%7D%0A%09%09%09%09%09%09%09++%7D [status]  200   [respoonse header]  map[Access-Control-Allow-Headers:[Authorization,content-type] Access-Control-Allow-Methods:[HEAD,GET,POST,OPTIONS] Access-Control-Allow-Origin:[*] Content-Type:[application/json] Date:[Fri, 23 Sep 2022 14:25:40 GMT]]
IMAGE NAME                        TAG                       DIGEST        SIZE      
busybox                           latest                    b9590ac9      773kB     
busybox                           new                       b9590ac9      773kB     
busybox                           1.0.1                     2ac88360      773kB     
golang                            1.18                      d45a1013      353MB     
hello-artifact                    v2                        74470033      12B       
python                            latest                    e8e1d47c      351MB     
test-regclient                    latest                    d66fe387      76MB      
test-regclient                    1.0.0                     d8be8fb7      76MB      
zot-chart/zot                     0.1.0                     cf8cd347      2.3kB     
zot-cve-test                      0.0.1                     98f02d9e      75MB      
zot-test                          0.0.1                     d66fe387      76MB    
 ```

**Review by andaaron — APPROVED** [review]
LGTM

**Review by shimish2 — APPROVED** [review]
lgtm

**Pull request merged**